### DISCUSSION TITLE
Publish NPM package

### DISCUSSION THREAD

**Discussion by juffalow** [community/community#25583] [Packages]
<p>Hi! I have Node.js <a href="https://github.com/juffalow/pentest-tool-lite" rel="nofollow noopener">project </a>and I want to publish it to npm registry when I create a new release on GitHub.</p>
<p>But when I try to do it with yarn, I’ve got this error:</p>
<p>No token found and can’t prompt for login when running with --non-interactive.</p>
<p>So I tried to do it with npm, but again error: “Unauthorized”.</p>
<p>And another thing: how can it now the next version? Can I somehow get it from GitHub release? I know I can updated it to next fix version, but what if I want to do minor / major update?</p>

**Comment by juffalow** [comment, marked as answer]
<p>If anyone would ancounter the same problem, here is the solution:</p>
<p>In $GITHUB_REF env variable is reference to the release version in format “refs/tags/&lt;tag&gt;”. You can remove the prefix and pass the version to publish command:</p>
<pre><code>run: yarn publish --new-version ${GITHUB_REF#"refs/tags/"}
</code></pre>
<p>It will publish the new version, but still ends with error. It is because it tries to also create a commit, but git needs your name and email to do it.</p>
<pre><code>Exit code: 128
Command: git
Arguments: commit -m v2.3.16
Directory: /home/runner/work/pentest-tool-lite/pentest-tool-lite
Output:
*** Please tell me who you are.

Run
  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for &lt;id&gt;) not allowed
</code></pre>
<p>You can either set the user email and name before the command or you can add --no-git-tag-version parameter for yarn publish command. The final publish command then looks like this:</p>
<pre><code>run: yarn publish --new-version ${GITHUB_REF#"refs/tags/"} --no-git-tag-version
</code></pre>

**Reply by KinNeko-De** [reply]
--no-git-tag-version works also if you use 'npm version'
You can create the git tag later with github.rest.git.createRef
